zeebe/restore/src/main/java/io/camunda/zeebe/restore/PartitionRestoreService.java

```diff
@@ -48,6 +51,14 @@ public PartitionRestoreService(final BackupStore backupStore, final RaftPartitio
     this.partition = partition;
   }
 
+  public PartitionRestoreService(
+      final BackupStore backupStore,
+      final RaftPartition partition,
+      final ChecksumProvider provider) {
+    this(backupStore, partition);
+    this.provider = provider;
+  }
+
```

Comment:
:wrench: These constructors are a bit confusing. I'd suggest to always have one "full" constructor which sets all the fields and then convenience constructors which all delegate to the full constructor and assign default values to some parameters.